internal/log_analysis/log_processor/parsers/pantherlog.go

```diff
@@ -126,14 +128,38 @@ func (pl *PantherLog) SetCoreFields(logType string, eventTime *timestamp.RFC3339
 	pl.PantherParseTime = &parseTime
 }
 
-// Will return true if the IP address was successfully appended, false if the value was not an IP
+// AppendAnyIPAddressPtr returns true if the IP address was successfully appended,
+// otherwise false if the value was not an IP
 func (pl *PantherLog) AppendAnyIPAddressPtr(value *string) bool {
 	if value == nil {
 		return false
 	}
 	return pl.AppendAnyIPAddress(*value)
 }
 
+// AppendAnyIPAddressInFieldPtr makes sure the value passed is not nil
+func (pl *PantherLog) AppendAnyIPAddressInFieldPtr(value *string) bool {
+	if value == nil {
+		return false
+	}
+	return pl.AppendAnyIPAddressInField(*value)
+}
+
+// AppendAnyIPAddressInField extracts all IPs from the value using a regexp
+func (pl *PantherLog) AppendAnyIPAddressInField(value string) bool {
+	matchedIPs := ipRegex.FindAll([]byte(value), -1)
+	if len(matchedIPs) > 0 {
+		if pl.PantherAnyIPAddresses == nil { // lazy create
+			pl.PantherAnyIPAddresses = NewPantherAnyString()
+		}
+		for _, match := range matchedIPs {
+			AppendAnyString(pl.PantherAnyIPAddresses, string(match))
+		}
+		return true
+	}
+	return false
+}
```

Comment:
nits: 
* Replace `FindAll` with `FindAllString` to avoid casting
* Can make the code more compact and avoid writing some logic that already exists by changing to something like below:
```suggestion
func (pl *PantherLog) AppendAnyIPAddressInField(value string) bool {
	result := false
	matchedIPs := ipRegex.FindAllString(value, -1)
	for _, match := range matchedIPs{
		result = result || pl.AppendAnyIPAddress(match)
	}
	return result
}
`
```